torch/distributions/distribution.py

```diff
@@ -196,13 +196,13 @@ def _validate_sample(self, value):
         and event shapes.
 
         Args:
-            value (Tensor): the tensor whose log probability is to be
+            value (tensor): the tensor whose log probability is to be
                 computed by the `log_prob` method.
         Raises
             ValueError: when the rightmost dimensions of `value` do not match the
                 distribution's batch and event shapes.
         """
-        if not isinstance(value, torch.Tensor):
+        if value.__class__.__name__ != 'Tensor':
```

Comment:
The check was correct before. `isinstance(value, torch.Tensor)` is the recommended way now

Reply:
I have reverted the change.

However, considering the deprecation of `torch.Tensor` based on the discussion in https://github.com/pytorch/pytorch/issues/7900 , how would you suggest this check be written?

Reply:
To be honest I didn't know of that thread, and I don't think there's any clear deprecation plan for `torch.Tensor` as a type (maybe there is as a tensor constructor).